datajoint/expression.py

```diff
@@ -119,17 +132,34 @@ def where_clause(self):
             else " WHERE (%s)" % ")AND(".join(str(s) for s in self.restriction)
         )
 
+    def sorting_clauses(self):
+        if self.top:
+            limit = self.top["limit"]
+            offset = self.top["offset"]
+            order_by = self.top["order_by"]
+        else:
+            return ""
+        if offset and limit is None:
+            raise DataJointError("limit is required when offset is set")
+        clause = ""
+        if order_by is not None:
```

Comment:
I am thinking that perhaps we should never use `LIMIT` without an `ORDER BY`. The default ordering should be by `KEY` rather than unordered. Let's discuss.

Reply:
If a limit is specified but no order, it will order by "KEY" by [default](https://github.com/datajoint/datajoint-python/pull/1084/files#diff-27f86ee6bd86a0f1f7ffaa9ae916330d6ec989e73ae0db9215801892ac30989cR74). However a user is still able to manually pass in `None` as a value, such as `dj.Top(None, None, 1)`. That is the edge case being handled here.

Reply:
In `dj.Top`, the input for `oder_by` does not allow the value `None`. I am okay with not allowing `None`. It simplifies the documentation burden and cognitive burden.  Since there is no performance hit for ordering by the primary key, there is no good reason to ever specify `order_by=None`.

Reply:
It is actually still possible to pass `None` as an argument
![image](https://github.com/datajoint/datajoint-python/assets/61991709/42a36ef6-23c0-45c5-bdaf-02040dabf8d4)

Reply:
yes, because Python ignores the type hinting.